include/seqan3/alignment/pairwise/detail/type_traits.hpp

```diff
@@ -80,6 +82,23 @@ template <typename configuration_t>
 //!\endcond
 struct alignment_configuration_traits
 {
+private:
+    //!\brief Helper function to determine the alignment result type.
+    static constexpr auto determine_alignment_result_type() noexcept
```

Comment:
Probably a stupid question, but why are we certain this is noexcept?

Reply:
because it is anyway just evaluated at compile time and within a constexpr context a function cannot throw. So this serves as an additional documentation, that we never expect something odd here.